src/main/java/org/mongoflink/bson/CdcDocument.java

```diff
@@ -0,0 +1,31 @@
+package org.mongoflink.bson;
+
+import org.bson.Document;
+
+import java.util.HashMap;
+
+/**
+ * CdcDocument is a bson Document with a delete flag used to indicate if a given Document in a
+ * DocumentBulk represents a mongo delete operation.
+ */
+public class CdcDocument extends Document {
```

Comment:
Could we use the decorator pattern to avoid creating a new Document? 

Something like:
```
public class CdcDocument extends Document {
    private Document inner;
    private boolean isDelete;
    ....
}
```

Reply:
are you suggesting to not invoke the `Document` super constructor?

Reply:
Not really. I mean perhaps we could leave the `CdcDocument` empty (invoke the empty constructor `super()`) and overwrite the methods to use the inner `Document` instead. 

However, I find it cumbersome and hard to maintain because `Document` has a lot of methods. Maybe we could try that approach if reordering becomes a problem. Now I'm good with the PR.